lib/datadog/appsec/component.rb

```diff
@@ -12,6 +12,7 @@ class Component
       class << self
         def build_appsec_component(settings, telemetry:)
           return unless settings.respond_to?(:appsec) && settings.appsec.enabled
+          return unless enable_appsec?(settings)
```

Comment:
I think the method with `?` is doing more than 1 thing. WDYT on splitting it on something like this

```suggestion
          return unless appsec_enabled?(settings)
          return if incompatible_ffi_version? 
```

Reply:
I removed `appsec_enabled?` method and added `incompatible_ffi_version?` here: [51a803f](https://github.com/DataDog/dd-trace-rb/pull/3969/commits/51a803f2b68f70da7b2eb9d49c8173501cb3d59d)